src/backend/tcop/postgres.c

```diff
@@ -4880,7 +4881,18 @@ PostgresMain(int argc, char *argv[], const char *username)
 						SetUserIdAndContext(cuid, false); /* Set current userid */
 
 					if (serializedIdentityLen > 0)
+					{
+						/*
+						 * serializedIdentity doesn't include '\0', which will cause core dump in SetupProcessIdentity() with using elog(DEBUG1).
+						 * So palloc a new string with '\0'.
+						 */
+						completeSerializedIdentity = (char *) palloc((serializedIdentityLen + 1) * sizeof(char) );
+						memcpy(completeSerializedIdentity, serializedIdentity, serializedIdentityLen);
+						completeSerializedIdentity[serializedIdentityLen] = '\0';
+						serializedIdentity = completeSerializedIdentity;
 						SetupProcessIdentity(serializedIdentity);
```

Comment:
As the MemoryContext which is MessageContext will reset every time, so it will not cause memory leak even it  errors.